Xamarin.Forms.Platform.Android/Forms.cs

```diff
@@ -191,54 +193,149 @@ static Color GetAccentColor()
 		class AndroidDeviceInfo : DeviceInfo
 		{
 			readonly IDeviceInfoProvider _formsActivity;
+			AndroidOrientationEventListener _androidOrientationEventListener;
+
 			readonly Size _pixelScreenSize;
 			readonly double _scalingFactor;
 
-			Orientation _previousOrientation = Orientation.Undefined;
+			bool _disposed;
 
 			public AndroidDeviceInfo(IDeviceInfoProvider formsActivity)
 			{
-				_formsActivity = formsActivity;
-				CheckOrientationChanged(_formsActivity.Resources.Configuration.Orientation);
-				formsActivity.ConfigurationChanged += ConfigurationChanged;
-
 				using (DisplayMetrics display = formsActivity.Resources.DisplayMetrics)
 				{
 					_scalingFactor = display.Density;
 					_pixelScreenSize = new Size(display.WidthPixels, display.HeightPixels);
-					ScaledScreenSize = new Size(_pixelScreenSize.Width / _scalingFactor, _pixelScreenSize.Width / _scalingFactor);
+					ScaledScreenSize = new Size(_pixelScreenSize.Width / _scalingFactor, _pixelScreenSize.Height / _scalingFactor);
 				}
+
+				// initialize screen orientation
+				_formsActivity = formsActivity;
+				SetScreenOrientation();
+				_formsActivity.ConfigurationChanged += OnConfigurationChanged;
 			}
 
-			public override Size PixelScreenSize
+			public override Size PixelScreenSize => _pixelScreenSize;
+
+			public override Size ScaledScreenSize { get; }
+
+			public override double ScalingFactor => _scalingFactor;
+
+			public override void BeginDeviceOrientationNotifications()
 			{
-				get { return _pixelScreenSize; }
+				if (_androidOrientationEventListener != null)
+					return;
+
+				_androidOrientationEventListener = new AndroidOrientationEventListener((Context)_formsActivity, SensorDelay.Normal);
+				_androidOrientationEventListener.OrientationChanged += OnDeviceOrientationChanged;
+
+				if (_androidOrientationEventListener.CanDetectOrientation())
+					_androidOrientationEventListener.Enable();
 			}
 
-			public override Size ScaledScreenSize { get; }
+			public override void EndDeviceOrientationNotifications()
+			{
+				if (_androidOrientationEventListener == null)
+				    return;
+
+				_androidOrientationEventListener.OrientationChanged -= OnDeviceOrientationChanged;
+				_androidOrientationEventListener.Disable();
+				_androidOrientationEventListener.Dispose();
+				_androidOrientationEventListener = null;
+			}
+
+			void OnDeviceOrientationChanged(object sender, int i)
+			{
+				SetDeviceOrientation(i);
+			}
+
+			void OnConfigurationChanged(object sender, EventArgs e)
+			{
+				SetScreenOrientation();
+			}
 
-			public override double ScalingFactor
+			void SetDeviceOrientation(int rotation)
 			{
-				get { return _scalingFactor; }
+				if (rotation == OrientationEventListener.OrientationUnknown)
+				{
+					DeviceOrientation = DeviceOrientation.Unknown;
+					return;
+				}
```

Comment:
There is no direct way on Android to detect face up/down changes. Indirectly, these can be detected with a `SensorManager` as explaind [here](http://www.ahotbrew.com/how-to-detect-forward-and-backward-tilt/), but this seems to be a lot of work for simple values. `rotation` is set to unknown when the device orientation is either face up or down.

On iOS and UWP, these values are supported natively.